### DISCUSSION TITLE
Error 430 with 8BIM clipping path

### DISCUSSION THREAD

**Discussion by HeinziTuberkel** [ImageMagick/ImageMagick#3598] [Help]
I use the ImageMagickObject COM+ to resize and trim images and convert them to png.
On the input side the images have several formats.
Some with transparency, some with Photoshop clipping path. 
To find out, if the picture has a clipping path, I use the parameters
```
<filename>
-format
%[8BIM:1999,2998:#1]
info:
```
In the returning string I only check if "<svg" is contained to determine whether to use the clipping path for trimming or transparency.

This works for most pictures, but now I have some pictures that return an error:
  **convert: 430: unable to write file `' @ error/constitute.c/WriteImage/1313:**  

I have no idea, where the **`'** comes from. The input filename was "D:\Temp\Test.psd". 
The same command works fine from the command line and returns a clipping path of almost 150kB and 3500 lines.

BTW: Is thee a way to check only, if there IS a clipping path and return true or false instead of receiving the complete clipping path?

TIA

**Comment by snibgo** [comment]
I don't know why you get an error with COM+. Perhaps it is writing to a buffer that is too small.

If you use "-clip" but there is no clipping path, IM raises an error. See https://imagemagick.org/script/command-line-options.php#clip
```
f:\web\im>%IMG7%magick toes.png -clip NULL:

magick: no clip path defined 'toes.png': No error @ error/image.c/ClipImagePath/730.
```
This shows that toes.png has no clipping path.

**Reply by HeinziTuberkel** [reply]
Hey, thanks, snigbo, good idea, shoulda thought of that ;-)
